play-testkit/src/test/java/play/grpc/testkit/PlayJavaFunctionalTest.java

```diff
@@ -61,15 +63,23 @@ public void returns404OnNonGrpcRequest() throws Exception {
   }
 
   @Test
-  public void returns500OnNonExistentGrpcMethod() throws Exception {
+  public void returnsGrpcUnimplementedOnNonExistentGrpcMethod() throws Exception {
     final WSResponse rsp = wsGet("/" + GreeterService.name + "/FooBar");
-    assertEquals(500, rsp.getStatus()); // Maybe should be a 426, see #396
+    assertEquals(200, rsp.getStatus());
+    assertEquals(
+      Integer.toString(Status.Code.UNIMPLEMENTED.value()),
+      rsp.getSingleHeader("grpc-status").get()
+    );
   }
 
   @Test
-  public void returns500OnEmptyRequestToAGrpcMethod() throws Exception {
+  public void returnsGrpcInternalErrorOnEmptyRequestToAGrpcMethod() throws Exception {
     final WSResponse rsp = wsGet("/" + GreeterService.name + "/SayHello");
-    assertEquals(500, rsp.getStatus()); // Maybe should be a 426, see #396
+    assertEquals(200, rsp.getStatus());
+    assertEquals(
+      Integer.toString(Status.Code.INTERNAL.value()),
```

Comment:
Nice. We should use `Status.Code.INTERNAL.value()` instead of `13` in `Specs2` and `scalatestplus-play` (see https://github.com/playframework/play-grpc/blob/12da1f1b70f3fc2e9ddfb43a585c94d0b37010f6/play-specs2/src/test/scala/play/grpc/specs2/PlaySpecs2Spec.scala#L49)

Happy to do it on a separate PR.